hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/ProtobufLogReader.java

```diff
@@ -410,8 +410,7 @@ protected boolean readNext(Entry entry) throws IOException {
       if (!walKey.hasFollowingKvCount() || 0 == walKey.getFollowingKvCount()) {
         LOG.trace("WALKey has no KVs that follow it; trying the next one. current offset={}",
           this.inputStream.getPos());
-        seekOnFs(originalPosition);
-        return false;
+        return true;
```

Comment:
For the core change, I wonder if it might still be possible to have replication delays with specific encryption as explained on HBASE-20604.

(though possible dataloss during WAL splitting is definitely much serious case to consider)

Reply:
In general I do not think a partial data can cause the protobuf message to have wrong field values, it will just fail the parsing of the WALKey message and we will return earlier. So if we can successfully parse the WALKey message, and the following kv count is 0, it means that the kv count is 0 when writting, it does not mean the kv data is not available yet...

If this is not the case, then I think it should be a serious bug of CryptoInputStream, as it may return incorrect bytes...

Reply:
Yeah that is quite reasonable.